### PR TITLE
[ProgressV2] Move expanded lessons and choice levels to redux.

### PR DESCRIPTION
**Pull request by lfryemason**
As part of the addition of a floating header, it will be much simpler to have expanded lessons and expanded choice levels in the redux store instead of in component state. This PR moves:
* `expandedChoiceLevels` from `ExpandedProgressDataColumn.jsx` to the redux store.
* `expandedLessonIds` from `SectionProgressV2.jsx` to the redux store.

All references to those structures was moved to redux. This means moving some from passed in props to redux references. Tests are also updated to put use redux.

## Links
[TEACH-940](https://codedotorg.atlassian.net/browse/TEACH-940)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx [hunk 1]
```diff
@@ -119,12 +98,7 @@ function ExpandedProgressDataColumn({
       <caption hidden={true}>
         {i18n.progressForLesson({lessonName: lesson.title})}
       </caption>
-      <ExpandedProgressColumnHeader
-        lesson={lesson}
-        removeExpandedLesson={removeExpandedLesson}
-        expandedChoiceLevels={expandedChoiceLevels}
-        toggleExpandedChoiceLevel={toggleExpandedChoiceLevel}
-      />
+      <ExpandedProgressColumnHeader lesson={lesson} />
```
Comment (lfryemason) [inline apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx:101]: This is a great example of why I wanted to do this. Dramatically reduces the number of props that are just passed through.

File: apps/src/templates/sectionProgress/sectionProgressRedux.js [hunk 2]
```diff
@@ -38,6 +50,32 @@ export const addDataByUnit = data => ({
   data,
 });
 
+export const loadExpandedLessonsFromLocalStorage = (scriptId, sectionId) => ({
+  type: LOAD_EXPANDED_LESSONS_FROM_LOCAL_STORAGE,
+  scriptId,
```
Comment (etaderhold) [inline apps/src/templates/sectionProgress/sectionProgressRedux.js]: Could we use `unit` rather than `script` more often throughout this code?

**Review by etaderhold — APPROVED** [review]
I like it! Does this have any performance implications?

**Comment by lfryemason** [comment]
> I like it! Does this have any performance implications?

This is actually a performance improvement because connecting to the redux store will prevent component re-renders unless the redux value changes. [SO thread](https://stackoverflow.com/questions/61283239/performance-connecting-to-the-redux-store-vs-passing-data-through-props)

For the most part this will be minimal, but there are a couple components that just passed the prop from parent to child that will see (hopefully) significantly fewer re-renders.

**Pull request merged**